src/bytes.rs

```diff
@@ -63,6 +79,16 @@ impl ByteBuf {
             bytes: Vec::with_capacity(cap)
         }
     }
+
+    pub fn as_vec(self) -> Vec<u8> {
+        self.bytes
+    }
+}
```

Comment:
what I mean is, you don't need to implement `Into`, it's generated, since `From` exists.